### PR TITLE
mono 6.10.0.104

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

---

Followup PR of #57787

### CODE REVIEW

**Review by alebcay — COMMENTED** [review]
I believe the revision on L39/40 also needs to be changed. I tried updating it per the instructions in the preceding comment but I could not get `msbuild` to build successfully. Also, the build might still need `python3` - maybe `depends_on "python@3" => :build` is still needed (not a runtime dependency as far as I can tell).

File: Formula/mono.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Mono < Formula
 
   depends_on "cmake" => :build
   depends_on "pkg-config" => :build
-  depends_on :macos # Due to Python 2
+  depends_on "python@3" => :build
```
Comment (SMillerDev) [inline Formula/mono.rb:15]: That's not a thing
```suggestion
  depends_on "python@3.8" => :build
```

**Comment by alebcay** [comment]
I'm impressed that this passed - especially since we didn't touch the `msbuild` revision (it broke for me on building when I _did_ change that revision).

**Comment by alebcay** [comment]
@SeekingMeaning could you please explain why you marked this `do not merge`? Is it because the `msbuild` revision was not updated, or is there some other issue?

**Comment by SeekingMeaning** [comment]
Does `msbuild` need to be updated?

**Comment by alebcay** [comment]
I believe it's supposed to be - based on the comment above that resource block, I would imagine that the two are pretty tightly coupled. I tried updating it locally and found that the updated `msbuild` would not build correctly (and yet apparently as seen here, the previous version of `msbuild` works all right).

**Comment by chenrui333** [comment]
```
Time Elapsed 00:02:24.27
##vso[task.setvariable variable=Artifacts;isSecret=false;isOutput=true]/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts
##vso[task.setvariable variable=Artifacts.Toolset;isSecret=false;isOutput=true]/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/toolset
##vso[task.setvariable variable=Artifacts.Log;isSecret=false;isOutput=true]/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/log/Release-MONO
##vso[task.setvariable variable=Temp;isSecret=false;isOutput=true]/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/tmp/Release-MONO
##vso[task.setvariable variable=TMP;isSecret=false;isOutput=true]/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/tmp/Release-MONO
Some command line switches were read from the auto-response file "MSBuild.rsp". To disable this file, use the "-noAutoResponse" switch.
/private/tmp/mono--msbuild-20200802-19633-mv2bjo/eng/../stage1/bin/bootstrap/net472/MSBuild/Current/Bin/MSBuild.dll /nologo /bl:/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/log/Release-MONO/ToolsetRestore.binlog /clp:Summary /clp:ErrorsOnly;NoSummary /m /nr:false /p:TreatWarningsAsErrors=true /p:ContinuousIntegrationBuild=true /p:__ToolsetLocationOutputFile=/private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/toolset/1.0.0-beta.20202.9.txt /t:__WriteToolsetLocation /v:minimal /warnaserror /private/tmp/mono--msbuild-20200802-19633-mv2bjo/artifacts/toolset/restore.proj
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
System.ComponentModel.Win32Exception (0x80004005): Cannot find the specified file
  at System.Diagnostics.Process.set_PriorityClass (System.Diagnostics.ProcessPriorityClass value) [0x00049] in <44a6c8127d0e4e2eb35f19123c6378be>:0 
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.set_PriorityClass(System.Diagnostics.ProcessPriorityClass)
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x001a4] in <4a7a183af9734381b5e67182876a604c>:0 

Unhandled Exception:
System.ComponentModel.Win32Exception (0x80004005): Cannot find the specified file
  at System.Diagnostics.Process.set_PriorityClass (System.Diagnostics.ProcessPriorityClass value) [0x00049] in <44a6c8127d0e4e2eb35f19123c6378be>:0 
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.set_PriorityClass(System.Diagnostics.ProcessPriorityClass)
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x0045b] in <4a7a183af9734381b5e67182876a604c>:0 
  at Microsoft.Build.CommandLine.MSBuildApp.Main () [0x00028] in <4a7a183af9734381b5e67182876a604c>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ComponentModel.Win32Exception (0x80004005): Cannot find the specified file
  at System.Diagnostics.Process.set_PriorityClass (System.Diagnostics.ProcessPriorityClass value) [0x00049] in <44a6c8127d0e4e2eb35f19123c6378be>:0 
  at (wrapper remoting-invoke-with-check) System.Diagnostics.Process.set_PriorityClass(System.Diagnostics.ProcessPriorityClass)
  at Microsoft.Build.CommandLine.MSBuildApp.Execute (System.String commandLine) [0x0045b] in <4a7a183af9734381b5e67182876a604c>:0 
  at Microsoft.Build.CommandLine.MSBuildApp.Main () [0x00028] in <4a7a183af9734381b5e67182876a604c>:0 
##vso[task.logissue type=error]Build failed (exit code '1').
```

**Comment by chenrui333** [comment]
Hi Folks (@SMillerDev, @alebcay, @SeekingMeaning ), what would be the direction moving forward? 

I am going to update this PR to the latest Mono 6.10 release.

**Comment by SMillerDev** [comment]
I'm not sure what kind of direction you're looking for.

**Comment by chenrui333** [comment]
yeah, I think we have two choices in here:
- ship out the current release (with outdated msbuild, as it passed the build/test)
- updated the release and msbuild release, 🤞 it would pass the build/test

**Comment by alebcay** [comment]
I'm going to pop into the Gitter chat for Mono and see if anyone can maybe provide some answers.

**Comment by chenrui333** [comment]
Thanks @alebcay!

**Comment by chenrui333** [comment]
I am gonna close this PR for now. We can revisit this later.

**Pull request closed without merging**